pkg/storegateway/bucket_stores.go

```diff
@@ -293,6 +300,11 @@ func (u *BucketStores) Series(req *storepb.SeriesRequest, srv storepb.Store_Seri
 	spanLog, spanCtx := spanlogger.New(srv.Context(), "BucketStores.Series")
 	defer spanLog.Span.Finish()
 
+	maxInflightRequests := u.cfg.BucketStore.MaxInflightRequests
+	if maxInflightRequests > 0 && u.getInflightRequestCnt() >= maxInflightRequests {
+		return ErrTooManyInflightRequests
+	}
```

Comment:
Let's move this check to L328 as well? No need to throw inflight request error as we might hit errors like L317

Reply:
Performance wise I don't think this makes much difference by checking inflight requests first. I expect users having CMK permission deny errors to always see permission deny error rather than the `ErrTooManyInflightRequests` one.

If we return `ErrTooManyInflightRequests` early, querier retries requests again, which might be more expensive and unnecessary. 

Let's move the limit check to the place where we actually know we are going to request SG.

Reply:
Sounds good, will do

Reply:
Done!